src/components/JurisdictionIdToName/index.tsx

```diff
@@ -0,0 +1,115 @@
+import store from '../../store';
+import jurisdictionsReducer, {
+  fetchGeoLessJurisdictions,
+  GeoLessJurisdiction,
+  getJurisdictionNamesByIds,
+  reducerName as JurisdictionReducerName,
+} from '../../store/ducks/jurisdictions';
+
+// register the jurisdiction reducer.
+reducerRegistry.register(JurisdictionReducerName, jurisdictionsReducer);
+
+/** interface for props for the JurisdictionIdsToNames component */
+interface Props {
+  id: string;
+  ids: string[];
+  serviceClass: typeof OpenSRPService;
+  idsToNamesLookup: FlexObject<string> | null;
+  fetchGeoLessJurisdictionsAction: typeof fetchGeoLessJurisdictions;
+}
+
+const defaultProps: Props = {
+  fetchGeoLessJurisdictionsAction: fetchGeoLessJurisdictions,
+  id: '',
+  ids: [],
+  idsToNamesLookup: null,
+  serviceClass: OpenSRPService,
+};
+
+/** The JurisdictionIdToName component take a jurisdictionId and renders
+ * the jurisdiction Name associated with that id
+ */
+const JurisdictionIdToName = (props: Props) => {
+  const { ids, serviceClass, idsToNamesLookup, id, fetchGeoLessJurisdictionsAction } = props;
+
+  /** async function that takes in the string ids and calls
+   * performs the api call and then updates the store
+   *
+   * @param {typeof OpenSRPService} service - the opensrp service class
+   * @param {string []} jurisdictionIds - a list of jurisdiction ids for which we shall
+   *    get geometry less jurisdictions.
+   * @param {} - action creator that returns action to add geometry less jurisdictions
+   */
+  async function loadJurisdictionNames(
+    service: typeof OpenSRPService,
+    jurisdictionIds: string[],
+    actionCreator: typeof fetchGeoLessJurisdictions
+  ) {
+    // perform api call and dispatch result
+    const serve = new service(OPENSRP_FIND_BY_JURISDICTION_ID);
+    const filterParams = {
+      jurisdiction_ids: jurisdictionIds,
+      return_geometry: false,
+    };
+    serve
+      .list(filterParams)
+      .then((response: GeoLessJurisdiction[]) => {
+        store.dispatch(actionCreator(response));
+      })
+      .catch((err: Error) => {
+        /** what are we doing with errors?? */
+      });
+  }
+
+  useEffect(() => {
+    // perform an api call only if the jurisdiction id is not in our store
+    if (!(idsToNamesLookup && idsToNamesLookup[id])) {
+      loadJurisdictionNames(serviceClass, ids, fetchGeoLessJurisdictionsAction);
+    }
+  }, []);
+
+  /** outputText will be jurisdiction name if jurisdiction id has truthy value
+   * form lookup else will have the jurisdiction id
+   */
+  const outputText = idsToNamesLookup && idsToNamesLookup[id] ? idsToNamesLookup[id] : id;
+
+  return (
+    <Fragment>
+      <span>{outputText}</span>
+    </Fragment>
+  );
+};
+
+JurisdictionIdToName.defaultProps = defaultProps;
+
+export { JurisdictionIdToName };
+
+// connect component
+
+/** interface for ampState returned obj */
+interface DispatchedProps {
+  idsToNamesLookup: FlexObject<string>;
```

Comment:
I think it is probably possible to not use FlexObject here - we already know what we are dealing with based on line 108, no?

Reply:
Then i think, i should remove the interface totally.
Ok, so if the types of the obj returned by `mapStateToProps` are implicitly well defined, then we dont need an interface describing the output, in this case the `DispatchedProps` interface?

Reply:
I had not considered not using it completely, I just noticed that we dont need to use FlexObject because we already know the shape of what we are dealing with